src/freertos_drivers/net_cc32xx/CC32xxWiFi.cxx

```diff
@@ -850,6 +863,52 @@ void CC32xxWiFi::sock_event_handler(void *context)
     }
 }
 
+/*
+ * CC32xxWiFi::http_server_callback()
+ */
+void CC32xxWiFi::http_server_callback(void *context1, void *context2)
+{
+    if(!context1 || !context2)
+    {
+        return;
+    }
+
+    SlHttpServerEvent_t *event = static_cast<SlHttpServerEvent_t *>(context1);
+    SlHttpServerResponse_t *response = static_cast<SlHttpServerResponse_t *>(context2);
+
+    switch (event->Event)
+    {
+        case SL_NETAPP_HTTPGETTOKENVALUE_EVENT:
+        {
+            unsigned char *ptr;
+
+            ptr = response->ResponseData.token_value.data;
+            response->ResponseData.token_value.len = 0;
+
+            for (unsigned i = 0; i < httpGetCallbacks_.size(); ++i)
+            {
+                if (memcmp(event->EventData.httpTokenName.data,
+                           httpGetCallbacks_[i].second,
+                           strlen(httpGetCallbacks_[i].second)) == 0)
+                {
+                    string result = httpGetCallbacks_[i].first();
+                    memcpy(ptr, result.c_str(), result.length());
```

Comment:
I've never used string::length() -- i always use size() for c++ container length. (We also use size() everywhere else in the codebase)